File: SPS.cpp
#include <iostream>
#include <conio.h>
#include <stdlib.h>
#include <ctime>
using namespace std;

int main()
{
    cout << "ABHISHEK GIRI WELCOMES YOU TO HIS GAME\n enjoyyyyyy....!!\n\n"
         << endl;
    string player_name;
    cout << "Enter your name: ";
    cin >> player_name;

    system("CLS");
    int rounds;
    cout << " \" " << player_name << " \" "
         << " how many rounds do you want to play: ";
    cin >> rounds;

    int player_score = 0, computer_score = 0;

    for (int round = 1; round <= rounds; round++)
    {
        system("CLS");
        int player_choice, computer_choice;
        cout << "Round no: " << round << "/" << rounds << "\n\n"
             << endl;

        cout << player_name << "'s Score = " << player_score << endl;
        cout << "computer's score = " << computer_score << "\n\n"
             << endl;

        cout << "Press 1 to choose stone" << endl;
        cout << "Press 2 to choose Paper" << endl;
        cout << "Press 3 to choose scissor" << endl;
        cout << "You chose: " << endl;
        do
        {
            cin >> player_choice;
        } while (player_choice != 1 && player_choice != 2 && player_choice != 3);

        srand(time(0));
        computer_choice = (rand() % 3) + 1;

        if (player_choice == 1 && computer_choice == 3)
        {
            cout << "Computer chose " << computer_choice << endl;
            cout << "You win" << endl;
            player_score++;
            cout << "Press Enter" << endl;
            getch();
        }
        else if (player_choice == 2 && computer_choice == 1)
        {
            cout << "Computer chose " << computer_choice << endl;
            cout << "You win" << endl;
            player_score++;
            cout << "Press Enter" << endl;
            getch();
        }
        else if (player_choice == 3 && computer_choice == 2)
        {
            cout << "Computer chose " << computer_choice << endl;
            cout << "You win" << endl;
            player_score++;
            cout << "Press Enter" << endl;
            getch();
        }
        else if (player_choice == computer_choice)
        {
            cout << "Computer chose " << computer_choice << endl;
            cout << "It's a draw" << endl;
            cout << "Press Enter" << endl;
            getch();
        }
        else
        {
            cout << "Computer chose " << computer_choice << endl;
            cout << "computer won" << endl;
            computer_score++;
            cout << "Press Enter" << endl;
            getch();
        }
    }
    cout << "\n"
         << player_name << " :Your score is " << player_score << " And Computers Score is " << computer_score << endl;

    cout << "\n\n Thank you for your time " << endl;
    cout << "from ABHISHEK GIRI";
    return 0;
}

File: SPS in Python/SPS.py
"""
1=> Rock
2=> Paper
3=> Scissors

"""
import random


def YourChoice():
    print('To Select "Rock" Press "1"')
    print('To Select "Paper" Press "2"')
    print('To Select "Scissor" Press "3"')

    YourChoice = int(input("Please enter your choice:- "))
    if(YourChoice == 1):
        print('You chose "Rock"')
        print("\n")

    elif(YourChoice == 2):
        print('You chose "Paper"')
        print("\n")

    elif(YourChoice == 3):
        print('You chose "Scissors"')
        print("\n")

    else:
        print("Invalid choice!! Choose again")
        print("\n")

    return YourChoice


def CompChoice():
    CompChoice = random.randrange(1, 3)
    if(CompChoice == 1):
        print('Computer chose "Rock"')
        print("\n")
    elif(CompChoice == 2):
        print('Computer chose "Paper"')
        print("\n")
    elif(CompChoice == 3):
        print('Computer chose "Scissors"')
        print("\n")

    return CompChoice


def winner(YourChoice, CompChoice):
    if(CompChoice == 1 and YourChoice == 2):
        print('YOU "WON"')
        print("\n")
        return 1

    elif(CompChoice == 2 and YourChoice == 3):
        print('YOU "WON"')
        print("\n")
        return 1

    elif(CompChoice == 3 and YourChoice == 1):
        print('YOU "WON"')
        print("\n")
        return 1

    elif(CompChoice == 1 and YourChoice == 3):
        print('COMPUTER "WON"')
        print("\n")
        return 0

    elif(CompChoice == 2 and YourChoice == 1):
        print('COMPUTER "WON"')
        print("\n")
        return 0

    elif(CompChoice == 3 and YourChoice == 2):
        print('COMPUTER "WON"')
        print("\n")
        return 0

    else:
        print('Its a "DRAW"')
        return 3
        print("\n")


name = input("Please enter your name:- \n")
print("\n")

rounds = int(input("How many rounds would you like to play? \n"))
print("\n")

YourScore = 0
ComputerScore = 0

for i in range(0, rounds):
    MYCHOICE = YourChoice()
    COMPUTERCHOICE = CompChoice()
    Score = winner(MYCHOICE, COMPUTERCHOICE)
    if(Score == 0):
        ComputerScore += 1
    elif(Score == 1):
        YourScore += 1

print("Your Total Score = " + str(YourScore) +
      " Computers Total Score = " + str(ComputerScore))
